### PR TITLE
Kv26 usb

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Comment by mike-scott** [comment]
@igoropaniuk Did you also want to include the slew rate changes from upstream?

File: meta-lmp-base/recipes-samples/images/lmp-image-common.inc
Comment (Tim-Anderson) [inline meta-lmp-base/recipes-samples/images/lmp-image-common.inc:39]: Is the compile tool needed for this?

**Comment by igoropaniuk** [comment]
@mike-scott done. the only problem is that dts overlays doesn't redefine usb1 nodes. Should that also be addressed here?

**Review by Tim-Anderson — APPROVED** [review]
Just the comment about dtc being included. Otherwise LGTM

**Comment by ricardosalveti** [comment]
Mind just moving the dtc addition into a separated pull request?

File: meta-lmp-base/recipes-samples/images/lmp-image-common.inc
Comment (mike-scott) [reply meta-lmp-base/recipes-samples/images/lmp-image-common.inc:39]: Not sure which compile tool you're referring to.  This tool is actually used to compile and decompile the device-tree binaries.

**Review by mike-scott — APPROVED** [review]
LGTM, leaving it up to @ricardosalveti on the DTC PR move.

**Review by quaresmajose — APPROVED** [review]
LGTM

**Comment by igoropaniuk** [comment]
Moved DTC patch to the separate PR

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-rev1.dts
Comment (mike-scott) [inline meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-rev1.dts:42]: I notice these are under the i2c node.  This driver expects the parent to be a USB node now, if I'm not mistaken?

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-revB.dts
Comment (mike-scott) [inline meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-revB.dts:42]: Same issue here for the i2c node

File: meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-rev1.dts
Comment (ricardosalveti) [reply meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-rev1.dts:42]: It works correctly with these changes only, so it might be ok.

Will merge as is to unblock this and we can update if needed.

Comment (MrCry0) [reply meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/kv260/zynqmp-sck-kv-g-rev1.dts:42]: > I notice these are under the i2c node. This driver expects the parent to be a USB node now, if I'm not mistaken?

@mike-scott These USB chips can use the I2C bus for management.

https://ww1.microchip.com/downloads/aemDocuments/documents/UNG/ProductDocuments/DataSheets/USB5744-Data-Sheet-DS00001855.pdf

The Kv260 seems to use this way to manage the USB chip.

**Pull request merged**
